src/core/folder.h

```diff
@@ -179,6 +183,8 @@ private Q_SLOTS:
     bool defer_content_test : 1;
 
     static std::unordered_map<FilePath, std::weak_ptr<Folder>, FilePathHash> cache_;
+    static FilePath cutFilesDirPath_;
+    static std::shared_ptr<const HashSet> cutFilesHashSet_;
```

Comment:
OK, I found what I was missing. `cutFilesHashSet_` should be static, of course, and `cutFilesDirPath_` is needed. The problem is that if another directory (like desktop) is loaded, `FolderModel::setCutFiles(const QItemSelection& selection)` won't work correctly because if that other directory calls it first, `refreshAll` will be made `false` for the folder in question.

Please tell me when you have time for a discussion! And sorry for my misunderstanding, which was caused by trying to find an easy solution to a problem!